newIDE/app/src/ObjectEditor/Editors/SpriteEditor/index.js

```diff
@@ -214,12 +214,29 @@ export default function SpriteEditor({
           }
         }
       }
-
       forceUpdate();
     },
     [spriteConfiguration, project, forceUpdate]
   );
 
+  const onApplyFirstSpriteCollisionMaskToSprite = React.useCallback(
+    (sprite: gdSprite) => {
+      if (spriteConfiguration.getAnimationsCount() === 0) return;
+      const firstAnimation = spriteConfiguration.getAnimation(0);
+      if (firstAnimation.getDirectionsCount() === 0) return;
+      const firstDirection = firstAnimation.getDirection(0);
+      if (firstDirection.getSpritesCount() === 0) return;
+      const firstSprite = firstDirection.getSprite(0);
+      const firstSpriteCollisionMask = firstSprite.getCustomCollisionMask();
```

Comment:
I'm not sure I understand, this will never be falsy?

Reply:
It is possible the first animation/direction/sprite is null, and the user is adding a sprite to a second animation.
In this case, we don't use the first one's size, we just set it to "fullImage"

Reply:
No I mean `firstSpriteCollisionMask`, it's an object returned by libGD.js, which are never null (they are pointers to the wasm memory that are wrapped by libGD.js into a JavaScript object, so they will never be null/undefined. And if the C++ returns an incorrect object, you will still have a valid JavaScript object, it will just crash the app when you try to use it)

So the next two lines (`!firstSpriteCollisionMask` and `if (firstSpriteCollisionMask)`) will always set false and the if will always pass.

Reply:
Oh right I get it now. I copied the logic from the matchingCollisionMask when it gets calculated (and it can fail)
But in this case, there's always one indeed. I'll change that to just copy the mask (as this is called only when automatic is set)